.github/workflows/pr-notification-slack.yaml

```diff
@@ -0,0 +1,338 @@
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+        run: |
+          echo "Condition value ------: ${{ steps.fetch_all_pull_requests_using_condition.outputs.notify }}"
+          pr_list=$(cat pr_list.txt)
+          payload=$(cat <<-EOF
+          {
+            "blocks": [
+              {
+                "type": "header",
+                "text": {
+                  "type": "plain_text",
+                  "text": "List of Open/New/Updated Pull Requests using Cron Job"
+                }
+              },
+              {
+                "type": "section",
+                "text": {
+                  "type": "mrkdwn",
+                  "text": "${pr_list}"
+                }
+              }
+            ]
+          }
+          EOF
+          )
+          echo "$payload"  # Debugging: Print payload to verify its format
+          curl -X POST -H 'Content-type: application/json' --data "$payload" $SLACK_WEBHOOK_URL || echo "Slack notification failed with status code: $?"
+
+      - name: Send Slack notification for closed PRs
+        if: success() && steps.fetch_closed_prs.outputs.notify_closed == 'true'
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+        run: |
+          echo "Condition value for closed PRs ------: ${{ steps.fetch_closed_prs.outputs.notify_closed }}"
+          closed_pr_list=$(cat closed_pr_list.txt)
+          payload=$(cat <<-EOF
+          {
+            "blocks": [
+              {
+                "type": "header",
+                "text": {
+                  "type": "plain_text",
+                  "text": "List of Pull Requests Closed Since the Last Cron Job"
+                }
+              },
+              {
+                "type": "section",
+                "text": {
+                  "type": "mrkdwn",
+                  "text": "${closed_pr_list}"
+                }
+              }
+            ]
+          }
+          EOF
+          )
+          echo "$payload"  # Debugging: Print payload to verify its format
+          curl -X POST -H 'Content-type: application/json' --data "$payload" $SLACK_WEBHOOK_URL || echo "Slack notification failed with status code: $?"
+
+      - name: Save current timestamp
+        run: |
+          current_timestamp=$(date -u +%Y-%m-%dT%H:%M:%SZ)
+          echo "$current_timestamp" > cache/last_run_timestamp.txt
+          echo "Current run timestamp: $current_timestamp"
+
+      - name: Verify Cache Save
+        if: always()
+        run: |
+          echo "Checking saved cache content...."
+          ls -l cache/
+          cat cache/notified_prs.json
+
+      - name: Save cache
+        if: always()
+        uses: actions/cache@v2
+        with:
+          path: cache
+          key: ${{ runner.os }}-pr-cache-${{ github.run_number }}
+
+      - name: Cleanup the restored cache key
+        run: |
+          gh extension install actions/gh-actions-cache
```

Comment:
We've already executed this step in 'Fetch recent cache'. Is it necessary to run it again here?

Reply:
No , I removed it from the code.